reframe/utility/os_ext.py

```diff
@@ -227,6 +227,11 @@ def inpath(entry, pathvar):
     return entry in set(pathvar.split(':'))
 
 
+def is_interactive():
+    '''Returns whether the given Python session is interactive'''
+    return bool(getattr(sys, 'ps1', sys.flags.interactive))
```

Comment:
In the extreme case that `sys.ps1 == ''`, this will fail. I suggest doing it as follows:

```suggestion
    return hasattr(sys, 'ps1') or sys.flags.interactive
```